docs/DeveloperGuide.md

```diff
@@ -1,29 +1,134 @@
+Please enter your new currency's full name: `bitcoin`
+Please enter your new currency's symbol: `bc`
+Please enter your new currency's rate: `0.000049`
+
+Based on that a new currency is created (in this example we have created a new currency of bitcoin).
+
+![avatar](/property/newcurrency2.png)
+
+Step 2: In order for the new currency to be added to the currencies file, firstly the equivalent string value of it is needed to write it. For that the `getPersonalCurrencyFileLine` method of the `PersonalCurrencyList` class is called. It returns a string in the format of `“!btc,bitcoin,bc,4.9E-6”`. You can notice that there is `‘!’` in front of the string, that indicates that it is a personal currency. Then the `writeToCurrencies` method of `NewCurrencyFileWorkings` class is called with the previously returned string value as parameter to write that line to the currencies file.
+
+
+Step 3: Then once it is successfully written to the `currencies.txt` file, then the previously created `Currency` is added both to the `currencyList` and to the `personalCurrencyList`. We need to keep track of both so it’s possible to remove that same currency from that `currencies.txt` file later.
+
+Step 4: Let’s say the user now wants to remove the previously created personal currency from the list of available currencies. It can type in the command `remove` and get prompted with this : 
+OK, which currency would you like to remove? Enter the abbreviation of that currency: `btc`
 
+Step 5: Now we have the abbreviation of the currency and we will add `‘!’` in front of it and create a new temporary currency object based on that with the `findCurrencyByAbbrName` method of the `CurrencyList` class and then check if the currencyList contains that currency. If so, then the `deleteFromCurrencies` method of the `NewCurrencyFileWorkings` class is called, which removes that currency from both of the currency lists and rewrites the `currencies.txt` file without that personal currency.
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+ Someone who works with many types of currencies, is interested in the forex market, or is interested in having a completely online wallet for tracking the money in the account.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+User who has a wallet which can have multiple currencies and can change them from one currency to another, essentially a bank account manager, where user can also withdraw and add money to the account. If they are trying to withdraw more money than they have in one currency we give them the option of exchange some from another currency (e.g trying to withdraw 100 SGD, but only have 80 SGD, we would tell them they can convert 14.34 EUR to match the remaining 20 SGD). They can give out requests to convert either one, select amount of currencies or all of the currencies they have to one specific currency. 
```

Comment:
You could perhaps expand on the currency conversion feature? Right now, it allows conversion rates only once, as far as I can tell. Does it allow updating the currency? Some clarification would be nice.